src/openrct2/world/Footpath.cpp

```diff
@@ -772,31 +769,29 @@ static bool footpath_disconnect_queue_from_path(int32_t x, int32_t y, TileElemen
  *  rct2: 0x006A6D7E
  */
 static void loc_6A6D7E(
-    int32_t initialX, int32_t initialY, int32_t z, int32_t direction, TileElement* initialTileElement, int32_t flags,
-    bool query, rct_neighbour_list* neighbourList)
+    const CoordsXYZD& pos, TileElement* initialTileElement, int32_t flags, bool query, rct_neighbour_list* neighbourList)
 {
-    int32_t x = initialX + CoordsDirectionDelta[direction].x;
-    int32_t y = initialY + CoordsDirectionDelta[direction].y;
-    if (((gScreenFlags & SCREEN_FLAGS_SCENARIO_EDITOR) || gCheatsSandboxMode) && map_is_edge({ x, y }))
+    auto shiftedPos = CoordsXY{ pos } + CoordsDirectionDelta[pos.direction];
```

Comment:
shiftedPos represents the coordinates of of the footpath/ride/entrance/mapEdge that is hopefully being connected to. `targetPos` might be a better name.